Change history and System Architectures (#997)
open-contracting/standard: docs/guidance/publish/change_history.md

The notice is represented in OCDS as a release with the 'planning' tag and data on the planned procurement, such as the estimated date the tender will be issued, is provided in the `tender` section with `.status` set to 'planned'. The `releases` list in the record for the contracting process is updated to include the ‘planning’ release, the only one that exists for the process at the moment. The compiled and versioned releases inside the record contain the latest information, which should be very similar to the 'planning' release. Compare the planning release and the record using the box below.

```eval_rst

.. jsoninclude:: ../../examples/planning.json
   :jsonpointer: /releases
   :expand: planning, tag, tender
   :title: release

```

```eval_rst

.. jsoninclude:: ../../examples/records/planning.json
   :jsonpointer: /records
   :expand: compiledRelease, planning, tender
   :title: record

```

### Tender

The London Borough of Barnet is ready to invite bids for the contract, so they issue the tender via an 'notice of intended procurement' (also known as 'contract notice' or 'opportunity notice').

> In OCDS, the notice is represented as a new release, using the 'tender' tag and providing details in the 'tender' section.

Reviewer: ```suggestion
In OCDS, the notice is represented as a new release, using the 'tender' tag and providing details in the `tender` section.
```

> No changes are made to the already published planning release, however the `tender` section of the new tender release is updated with the actual date the tender was issued, along with additional information about the deadline for bid submission, the items being procured, links to copies of tender documents and the estimated tender value.

Reviewer: ```suggestion
No changes are made to the already published planning release, however the `tender` section of the new tender release is updated with the actual date the tender was issued and new information is provided about the deadline for bid submission, the items being procured, links to copies of tender documents and the estimated tender value.
```

The tender release is added to the `releases` list in the record for the contracting process and the compiled release is updated to reflect the latest values for the new and updated fields in the 'tender' release. Note that the bid submission dates have been updated in the compiled release, and the versioned release now contains a list of changes for each field.

```eval_rst

.. jsoninclude:: ../../examples/tender.json
   :jsonpointer: /releases
   :expand: tender, tag, documents
   :title: release

```

```eval_rst

.. jsoninclude:: ../../examples/records/tender.json
   :jsonpointer: /records
   :expand: compiledRelease, tender, documents
   :title: record

```

### Tender Update

The enquiry period has ended, and a few questions from potential suppliers have been received. The procuring entity issues a document with the responses to the enquiries received from bidders.

In OCDS, a new release is published using the 'tenderUpdate' tag.

No changes are made to the existing planning and tender releases, however in the new release a link to the enquiry responses is added to `tender.documents`.

> The `hasEnquiries` field is set to true.

Reviewer: ```suggestion
No changes are made to the existing planning and tender releases, however in the new release a link to the enquiry responses is added to `tender.documents` and the `hasEnquiries` field is set to true.
```

> The publisher has decided to include all unchanged fields and values for the previous immutable release in the new one.

Reviewer: ```suggestion
The publisher has decided to repeat fields and values which are unchanged from the previous releases in the new one.
```

> The record for the contracting process now has three immutable releases, and updated compiled and versioned releases. Note that the compiled release has included the enquiries document in the tender section, and the field hasEnquiries has more than one entry in the versioned release.

Reviewer: ```suggestion
The record for the contracting process now has three immutable releases, and updated compiled and versioned releases. Note that the compiled release includes the enquiries document in the tender section, and the field `tender.hasEnquiries` has more than one entry in the versioned release.
```

```eval_rst

.. jsoninclude:: ../../examples/tenderUpdate.json
   :jsonpointer: /releases
   :expand: tender, tag, documents
   :title: release

```

```eval_rst

.. jsoninclude:: ../../examples/records/tenderUpdate.json
   :jsonpointer: /records
   :expand: compiledRelease, tender, documents
   :title: record

```

### Award

The procuring entity makes the decision to award the contract to Balfour Beatty and issues an award notice. 

The award notice is represented as a new OCDS release with the details of the award provided in the `award` section and using the 'award' tag. An entry is added to the `parties` array to provide information about the supplier and the full tender block is included as well with the `.status` field set to 'complete'. No changes are made to the existing planning, tender and tenderUpdate releases.

> The record for the contracting process has been updated to include the latest release, and the compiled and versioned releases have been updated accordingly. The compiled release has an award block and the supplier in parties, and the status of the tender has been updated to 'complete' as well. The versioned release also reflects the change in the status field.

Reviewer: ```suggestion
The record for the contracting process has been updated to include the latest release, and the compiled and versioned releases have been updated accordingly. The compiled release has an award block and an entry for the supplier in the `parties` section. The status of the tender has been updated to 'complete'. The versioned release also reflects the change in the `tender.status` field.
```

```eval_rst

.. jsoninclude:: ../../examples/award.json
   :jsonpointer: /releases
   :expand: parties, tender, awards, value, suppliers, contractPeriod
   :title: release

```

```eval_rst

.. jsoninclude:: ../../examples/records/award.json
   :jsonpointer: /records
   :expand: compiledRelease, parties, tender, awards, value, suppliers, contractPeriod
   :title: record

```

### Contract

After the standstill period has passed and no complaints have been issued against the award of the contract, the procuring entity and the supplier sign the contract. A new OCDS release is published using the 'contract' tag with the `contracts` section populated.

The record now has an additional release, and the compiled and versioned releases now include the contract section with no other changes.

```eval_rst

.. jsoninclude:: ../../examples/contract.json
   :jsonpointer: /releases
   :expand: contracts, value, period
   :title: release

```

```eval_rst

.. jsoninclude:: ../../examples/records/contract.json
   :jsonpointer: /records
   :expand: compiledRelease, contracts, value, period
   :title: record

```

### Implementation

> The supplier has started the construction work, and the procuring entity makes the first payment against the contract using their finance system, which shares a common contract identifier with the contract register. 

Reviewer: ```suggestion
The supplier has started the construction work, and the procuring entity makes the first payment against the contract using their finance system, which shares a common contract identifier with the procurement system used to publish the earlier stages of the contracting process. 
```

An OCDS release is published from the finance system using the 'implementation' tag, with details of the payment provided in the `contracts.implementation.transactions` section.

> Since the previous releases from the contract register share a common `ocid` with the release from the finance system, it’s possible to update the OCDS record for the contracting process with the new information from the finance system.

Reviewer: ```suggestion
Since the previous releases from the procurement system share a common `ocid` with the release from the finance system, it’s possible to update the OCDS record for the contracting process with the new information from the finance system.
```

<div class="example hint" markdown=1>

<p class="first admonition-title">Combining data from different systems</p>

Refer to the guidance on system architecture for more information on how OCDS can be used to combine data from different systems.

</div>

The new release from the finance system is added to the releases list in the OCDS record and both the compiled and versioned releases are updated to include the new transaction.

```eval_rst

.. jsoninclude:: ../../examples/implementation.json
   :jsonpointer: /releases
   :expand: contracts, implementation, transactions
   :title: release

```

```eval_rst

.. jsoninclude:: ../../examples/records/implementation.json
   :jsonpointer: /records
   :expand: compiledRelease, contracts, implementation, transactions
   :title: record

```

### Contract Amendment

> Due to unforeseen complications with subsistence in the area of the works, the procuring entity and the supplier agree to a contract extension to cover the additional time and cost required to complete the works. The contract register is updated with the new contract value and period and a modification notice is issued, this is represented in OCDS as a new release with the 'contractAmendment' tag.

Reviewer: ```suggestion
Due to unforeseen complications with subsistence in the area of the works, the procuring entity and the supplier agree to a contract extension to cover the additional time and cost required to complete the works. The procurement system is updated with the new contract value and period and a modification notice is issued, this is represented in OCDS as a new release with the 'contractAmendment' tag.
```

# Change history

> The following example shows how to publish a contracting process' data as it unfolds. Releases are built and published for each event in the process. Records are updated for each new release available.

Reviewer: ```suggestion
The following example shows how to publish data about a contracting process. Publish a release for each event in the process and update the record each time you publish a release.
```

Each subsection refers to a stage or event in the contracting process lifetime.

Refer to the [change history](../../getting_started/change_history) guidance for an introduction to the concepts involved.

### Planning

> The London Borough of Barnet has a planned tender for later in the year. To prepare the market they publish a *notice of planning procurement*. This is also known as *prior information notice*, or *future opportunity notice*.

Reviewer: ```suggestion
The London Borough of Barnet plans to publish a tender for cycle lane improvements later in the year. To prepare the market they publish a *notice of planned procurement*. This is also known as *prior information notice*, or *future opportunity notice*.
```

> The publisher creates an OCDS release to represent this notice. The release uses the 'planning' tag and includes data of the planned procurement. The `tender` section holds the estimated values, as the issue date. The section also uses the 'planned' status to state these are planned values.

Reviewer: ```suggestion
The publisher creates an OCDS release to represent this notice. The release uses the 'planning' tag and includes data about the planned procurement. The `tender` section holds the estimated values and publication date. The tender status is 'planned'.
```

> The publisher also creates an OCDS Record for the starting process. The releases list includes the new and only release so far. The compiled and versioned releases are also created for the record. Since there is only one release, the compiled version should look very much like this release. Compare the planning release and the record using the box below.

Reviewer: ```suggestion
The publisher also creates an OCDS record for the new contracting process. The releases list includes the new and only release so far. The compiled and versioned releases are also created for the record. Since there is only one release, the compiled version should look very much like this release. Compare the planning release and the record using the box below.
```

### Tender

The London Borough of Barnet is ready to invite bids for the contract. They issue the tender via an *notice of intended procurement*. This is also known as *contract notice* or *opportunity notice*.

The publisher creates a new release to represent this notice. The release uses the 'tender' tag and provides the details in the `tender` section.

> There are no changes to the planning release published before. But the new release has the updated information about the tender. This includes the real issue date, and the following new information:

Reviewer: ```suggestion
There are no changes to the planning release published before. But the new release has the updated information about the tender. This includes the actual publication date, and the following new information:
```

### Implementation

The supplier starts the construction work. After a while, the procuring entity makes the first payment to the supplier. The publisher creates a release to document this update in the process.

Up to this point, the publisher used a single source to generate OCDS documents. They store procurement data in a procurement system. Payment data, however, gets stored in a financial system, so do we need to join the data to create a release?

> If the systems share common *process identifiers*, there is no need to join data to create a release. The only requisite is to built the `ocid` using the shared identifier.

Reviewer: I think the mention of joining data might cause confusion, so I tried to find alternative language:

```markdown
So far, the council used a single procurement system to manage the process. The procurement system published the previous OCDS releases.

The council now uses a separate financial system to manage payments. The financial system publishes the new OCDS release.

The procurement system and the financial system share a common contracting process identifier. This means that the two systems can publish releases using the same `ocid`.
```

The publisher uses the information from the financial system to create a new release. This release uses the 'implementation' tag. The `contracts.implementation.transactions` section includes the details of the payment.

> The publisher produces the same `ocid` for the process using the common identifier. This way, is possible to update the record with the new release.

Reviewer: ```suggestion
Because the new release uses the same `ocid`, it is possible to update the record with the information.
```
